sql/binlog.cc

```diff
@@ -112,13 +112,24 @@ static ulonglong binlog_global_snapshot_position;
 // Binlog position variables for SHOW STATUS
 static char binlog_snapshot_file[FN_REFLEN];
 static ulonglong binlog_snapshot_position;
+static char *binlog_snapshot_gtid_executed= 0;
+static size_t binlog_snapshot_gtid_executed_len= 0;
+
+static const char *snapshot_gtid_executed_var_name= "snapshot_gtid_executed";
+// Fallback value in case of failed allocation. Very unlikely, but...
+static const char *binlog_snapshot_gtid_executed_fb= "";
 
 static SHOW_VAR binlog_status_vars_detail[]=
 {
   {"snapshot_file",
    (char *)&binlog_snapshot_file, SHOW_CHAR, SHOW_SCOPE_GLOBAL},
   {"snapshot_position",
    (char *)&binlog_snapshot_position, SHOW_LONGLONG, SHOW_SCOPE_GLOBAL},
+   /* Structure needs to be initialized with some value pointer (null here).
+   It will be updated to the current, dynamically allocated one
+   in set_binlog_snapshot_gtid_executed() */
+  {snapshot_gtid_executed_var_name, binlog_snapshot_gtid_executed, SHOW_CHAR,
```

Comment:
@kamil-holubicki Digging a bit deeper I encountered that there is also SHOW_FUNC kind of SHOW_VAR, that in my opinion is more suitable here.
Check `caching_sha2_password_status_variables` in `sha2_password.cc` as an example.
In this case, you will be able to have global std::string object (as in the previous iteration) and all manual memory management should go away.
Moreover, you won't need `get_var_index()` in this case

Reply:
Done.